hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HeapMemoryManager.java

```diff
@@ -317,12 +317,15 @@ private void tune() {
       unblockedFlushCnt = unblockedFlushCount.getAndSet(0);
       tunerContext.setUnblockedFlushCount(unblockedFlushCnt);
       metricsHeapMemoryManager.updateUnblockedFlushCount(unblockedFlushCnt);
-      // TODO : add support for offheap metrics
       tunerContext.setCurBlockCacheUsed((float) blockCache.getCurrentSize() / maxHeapSize);
       metricsHeapMemoryManager.setCurBlockCacheSizeGauge(blockCache.getCurrentSize());
+      long globalMemstoreDataSize = regionServerAccounting.getGlobalMemStoreDataSize();
       long globalMemstoreHeapSize = regionServerAccounting.getGlobalMemStoreHeapSize();
+      long globalMemStoreOffHeapSize = regionServerAccounting.getGlobalMemStoreOffHeapSize();
       tunerContext.setCurMemStoreUsed((float) globalMemstoreHeapSize / maxHeapSize);
-      metricsHeapMemoryManager.setCurMemStoreSizeGauge(globalMemstoreHeapSize);
```

Comment:
So in the old time we just use heap size as memstore size? This should be bug?

Reply:
yes, this is what i was wondering.
FYI @bbeaudreault as he discovered this

Reply:
Ah interesting. 

So I wasn't talking about these metrics at all. These metrics show up in JMX under `sub=Memory` (which for us is all 0's for some reason, and we don't use them). 

The metrics I was referring to are in `sub=Server` and also per-region and per-table metrics. These also have a `memStoreSize` field, and it is derived from the DataSize rather than HeapSize. These are calculated in a few places, you have to sort of dig in based on usages of `MetricsRegionServerSource.MEMSTORE_SIZE`

Reply:
Sorry for the confusion, I didn't realize these metrics existed. Maybe they are better? I need to read more.

Reply:
i was not aware either, realized these metrics exist and this is likely bug only after Jing created this PR, i also need to do some digging here.

Thanks for pointing out `sub=Server` @bbeaudreault

FYI @jinggou

Reply:
@bbeaudreault Sorry I didn't realize that you are referring to `sub=Server`. This change is made for `sub=Memory`, so I create a new pr (https://github.com/apache/hbase/pull/5308) for HBASE-27892.

For the code change here, I think it might be a bug because memStoreSize should refer to memstore data size instead of memstore heap size. I've created a new jira HBASE-27948 to figure out this issue. @Apache9 @virajjasani